### PR TITLE
Update various tools

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
Update various tools used in gardener.
Maintain kubernetes versions used in local setup.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: example/provider-local/garden/base/cloudprofile.yaml [hunk 1]
```diff
@@ -8,6 +8,12 @@ spec:
   - name: local
   kubernetes:
     versions:
+    - version: 1.28.4
+    - version: 1.27.8
+    - version: 1.26.11
+    - version: 1.25.16
+    - version: 1.24.17
+    # TODO(shafeeqes): Drop old versions once gardener v1.86 is released.
```
Comment (acumino) [inline example/provider-local/garden/base/cloudprofile.yaml]: IMO this is not worth doing. It doesn't provide any value.
```suggestion
```

**Review by acumino — CHANGES_REQUESTED** [review]

File: example/provider-local/garden/base/cloudprofile.yaml (hunk 1 above)
Comment (shafeeqes) [reply example/provider-local/garden/base/cloudprofile.yaml]: Isn't it always better to use the latest version with all the patches? I am not suggesting to always maintain this, but does it hurt?

Comment (shafeeqes) [reply example/provider-local/garden/base/cloudprofile.yaml]: No hard opinions though.  I can remove it if you want.

Comment (acumino) [reply example/provider-local/garden/base/cloudprofile.yaml]: IMO we don't need to create a two-release cycle just for this. k8s patch version doesn't provide much value for e2e. e2e test is more about testing gardener flow.

**Review by acumino — COMMENTED** [review]
Thanks!
/lgtm
/approve

**Pull request merged**